### PR TITLE
Fix slice of slice bug

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?
Weird corner case found with slicing vectors. This fixes that with tests added by @eraly 
(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

Please review
https://github.com/deeplearning4j/deeplearning4j/blob/master/CONTRIBUTING.md before opening a pull request.

### CODE REVIEW

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ndarray/BaseNDArray.java [hunk 1]
```diff
@@ -6052,6 +6053,12 @@ public INDArray convertToDoubles() {
             return DataBuffer.TypeEx.FLOAT;
         } else if (type == DataBuffer.Type.DOUBLE) {
             return DataBuffer.TypeEx.DOUBLE;
+
+        } else if(type == DataBuffer.Type.INT) {
+            return DataBuffer.TypeEx.INT8;
+        } else if(type == DataBuffer.Type.LONG) {
+            return DataBuffer.TypeEx.INT16;
```
Comment (raver119) [inline nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ndarray/BaseNDArray.java]: Nono. INT should be INT32. And LONG should be INT64 if there's INT64.

Comment (raver119) [reply nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ndarray/BaseNDArray.java]: i mean number in INT8 stands for number of bits :)
